src/aabb.rs

```diff
@@ -406,13 +406,17 @@ impl<S: BaseFloat> Continuous<Aabb2<S>> for Ray2<S> {
             let tx2 = (aabb.max.x - ray.origin.x) / ray.direction.x;
             tmin = tmin.max(tx1.min(tx2));
             tmax = tmax.min(tx1.max(tx2));
+        } else if ray.origin.x <= aabb.min.x || ray.origin.x >= aabb.max.x {
```

Comment:
we need to properly define the boundary semantics here for AABB, e.g. both min/max are included, or both excluded, and so on. Pretty sure checking for `<=` and `>=` is inconsistent with other checks w.r.t bounds. For example, we compare `t` without equality in `tmin < S::zero()`